main.js

```diff
@@ -0,0 +1,275 @@
+/******CONSTANTS*******/
+const Medium = document.getElementById("Medium");
+const Hard = document.getElementById("Hard");
+const Easy = document.getElementById("Easy");
+const highScore = document.getElementById("high-score");
+const Score = document.getElementById("score");
+const speedRadios = document.querySelectorAll('input[name="speed"]');
+
+const canvas = getCanvas();
+const ctx = canvas.getContext("2d");
+let newDirection = "right", oldDirection = "right";
+let v =0;// 
+let food = {x:300,y:300}
+let highscore=0;
+
+let body = [{ x: 300, y: 300, newDirection: "right"}, { x: 270, y: 300, newDirection: "right"}, { x: 240, y: 300, newDirection: "right"}, { x: 210, y: 300, newDirection: "right"}];
+
+
+function getCanvas() {
+    // Select the game canvas
+    const canvas = document.querySelector("#game-canvas");
+    return canvas;
+}
+
+function setCanvasSize() {
+    const parent = canvas.parentNode;
+    canvas.width = parent.offsetWidth;
+    while(canvas.width % 30 !=0){
+        canvas.width++;
+    }
+    canvas.height = parent.offsetHeight;
+    while(canvas.height % 30 !=0){
+        canvas.height++;
+    }
```

Comment:
Please use math instead of a while loop.Example:
```ts
// BAD CODE
 while(canvas.width % 30 !=0){
        canvas.width++;
   }
// BETTER CODE
canvas.width += canvas.width % 30;
```